c2corg_ui/static/js/search/searchfilters.js

```diff
@@ -189,4 +223,42 @@ app.SearchFiltersController.prototype.selectOption = function(prop, val, event)
 };
 
 
+/**
+ * @export
+ */
+app.SearchFiltersController.prototype.clear = function() {
+  for (var key in this.filters) {
+    this.location_.deleteParam(key);
+  }
+  this.filters = {};
+  this.orientations = [];
+  this.scope_.$root.$emit('searchFilterClear');
+};
+
+
+/**
+ * @param {string} orientation
+ * @param {app.SearchFiltersController} ctrl
+ * @param {goog.events.Event | jQuery.Event} e
+ * @param {string} filterName Name of the filter param in the URL.
+ * @export
+ */
+app.SearchFiltersController.prototype.setOrientation = function(orientation,
+    ctrl, e, filterName) {
+  if (this.orientations.indexOf(orientation) === -1) {
+    this.orientations.push(orientation);
+  } else {
+    this.orientations = this.orientations.filter(function(val) {
+      return val !== orientation;
+    });
```

Comment:
I don't understand what you are doing with the filter. You are creating a new array that contains all orientations in `this.orientations` that are not `orientation`. But in the `if` above you have already checked that `orientation` is not in `this.orientations`?